core/ops.rs

```diff
@@ -20,19 +22,83 @@ use std::ops::DerefMut;
 use std::pin::Pin;
 use std::rc::Rc;
 
-pub type OpAsyncFuture = Pin<Box<dyn Future<Output = Box<[u8]>>>>;
-pub type OpFn = dyn Fn(Rc<RefCell<OpState>>, BufVec) -> Op + 'static;
+pub use erased_serde::Serialize as Serializable;
+pub type RcOpState = Rc<RefCell<OpState>>;
+pub type PromiseId = u64;
+pub type OpBuf = Option<ZeroCopyBuf>;
+pub type OpAsyncFuture = Pin<Box<dyn Future<Output = OpResponse>>>;
+pub type OpFn = dyn Fn(RcOpState, OpPayload, OpBuf) -> Op + 'static;
 pub type OpId = usize;
 
+pub struct OpPayload<'a, 'b, 'c> {
+  pub(crate) scope: Option<&'a mut v8::HandleScope<'b>>,
```

Comment:
Interesting idea, wrapping up a scope and value together in a struct.
I hope it is actually safe.

Reply:
Can you elaborate on why you think it might be unsafe ?

Reply:
**IF** this is going on:

- create HandleScope A
- create HandleScope B
- allocate Local in HandleScope A
- anything happens with HandleScope B (drop?)

In other words, they "cross" each other, that'd be unsafe (a crash, most likely).
Looks like it doesn't happen in this PR, but only you know the fine print of your code.

A bit of background: `HandleScope`s are RAII objects in C++ and they cannot be moved.
They also shadow each other - an inner `HandleScope` inhibits the creation of a Local handle associated with the outer `HandleScope`. But `Local` handles that already exist in the outer `HandleScope` are safely accessible at all times.

These semantics are pretty much impossible to express in the Rust type system, so we do some extreme acrobatics (https://github.com/denoland/rusty_v8/blob/main/src/scope.rs) to make most calls involving `HandleScope` and other scopes safe, and in the end I *think* we managed. 

But moving a HandleScope into a struct and unloading it later is something I believe we haven't specifically covered, so I am not 100% sure there are no edge cases where the compiler would let you do something that you shouldn't be doing (here's what we have tests for: https://github.com/denoland/rusty_v8/tree/main/tests/compile_fail)